torchgeo/datamodules/vaihingen.py

```diff
@@ -24,26 +29,46 @@ class Vaihingen2DDataModule(pl.LightningDataModule):
 
     def __init__(
         self,
-        batch_size: int = 64,
+        train_batch_size: int = 32,
         num_workers: int = 0,
+        patch_size: Union[Tuple[int, int], int] = (64, 64),
+        num_patches_per_tile: int = 32,
         val_split_pct: float = 0.2,
         **kwargs: Any,
     ) -> None:
         """Initialize a LightningDataModule for Vaihingen2D based DataLoaders.
 
         Args:
-            batch_size: The batch size to use in all created DataLoaders
+            train_batch_size: The batch size used in the train DataLoader
+                (val_batch_size == test_batch_size == 1). The effective batch size
```

Comment:
Why does val/test have to have a batch size of 1?

Reply:
That's how it is implemented in the OSCD module. I guess the idea was to have random crops during training but then evaluate on the entire uncropped image during val/test?

Reply:
yes